kotlinx-coroutines-core/common/src/channels/Broadcast.kt

```diff
@@ -28,7 +29,12 @@ fun <E> ReceiveChannel<E>.broadcast(
 ): BroadcastChannel<E> =
     GlobalScope.broadcast(Dispatchers.Unconfined, capacity = capacity, start = start, onCompletion = consumes()) {
         for (e in this@broadcast) {
-            send(e)
+            try {
+                send(e)
+            } catch (e: ClosedSendChannelException) {
```

Comment:
Please add a test that shows the necessity of this `catch` block. It is not clear it is really required